sql-plugin/src/main/scala/org/apache/spark/sql/rapids/aggregate/aggregateFunctions.scala

```diff
@@ -287,12 +290,13 @@ class CudfMergeM2 extends CudfAggregate {
       }
     }
 
-  override lazy val groupByAggregate: GroupByAggregation = GroupByAggregation.mergeM2()
+  override lazy val groupByAggregate: GroupByAggregation =
+    GroupByAggregation.hostUDF(new CentralMomentHostUDF(AggregationType.GroupByMerge))
```

Comment:
Do we need a more specific type (or name) for this M2 aggregation, e.g. GroupByMergeM2 ?

Reply:
There will be just a few types: groupby/reduction and their merge variants, similar to this: https://github.com/NVIDIA/spark-rapids-jni/blob/branch-25.06/src/main/java/com/nvidia/spark/rapids/jni/HyperLogLogPlusPlusHostUDF.java#L66

I plan to extract `AggregationType` into a common data structure and use it across all host_udf classes.

Reply:
Note that `M2` and `MergeM2` will be deprecated and replaced by `CentralMoment` to be more consistent with Spark.

Reply:
Sorry I changed my mind. I'll implement the fix in libcudf instead, so we will have less duplicate code.